Performance plugin failed to start

```
2019-07-06 12:41:45 [pool-2-thread-1] WARN  n.r.c.plugins.config.ConfigPanel - Error when starting plugin FpsPlugin
net.runelite.client.plugins.PluginInstantiationException: java.lang.reflect.InvocationTargetException
	at net.runelite.client.plugins.PluginManager.startPlugin(PluginManager.java:377)
	at net.runelite.client.plugins.config.ConfigPanel.lambda$startPlugin$35(ConfigPanel.java:1215)
	at net.runelite.client.util.RunnableExceptionLogger.run(RunnableExceptionLogger.java:41)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException: null
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1349)
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1324)
	at javax.swing.SwingUtilities.invokeAndWait(SwingUtilities.java:1353)
	at net.runelite.client.plugins.PluginManager.startPlugin(PluginManager.java:349)
	... 9 common frames omitted
Caused by: java.lang.RuntimeException: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@56a4396 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@2cc3e0ce[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 78]
	at net.runelite.client.plugins.PluginManager.lambda$startPlugin$7(PluginManager.java:357)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@56a4396 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@2cc3e0ce[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 78]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
	at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:326)
	at java.util.concurrent.ScheduledThreadPoolExecutor.scheduleAtFixedRate(ScheduledThreadPoolExecutor.java:573)
	at java.util.concurrent.Executors$DelegatedScheduledExecutorService.scheduleAtFixedRate(Executors.java:735)
	at net.runelite.client.util.ExecutorServiceExceptionLogger.scheduleAtFixedRate(ExecutorServiceExceptionLogger.java:153)
	at net.runelite.client.plugins.fps.FpsPlugin.startUp(FpsPlugin.java:140)
	at net.runelite.client.plugins.PluginManager.lambda$startPlugin$7(PluginManager.java:353)
	... 14 common frames omitted
```

This error is thrown when trying to turn on the Performance plugin with Limit mode set to Unfocused and will not limit the FPS when unfocused. This isn't an issue when using Vanilla RuneLite

1.5.29-SNAPSHOT

--- a/runelite-client/src/main/java/net/runelite/client/plugins/fps/FpsPlugin.java
+++ b/runelite-client/src/main/java/net/runelite/client/plugins/fps/FpsPlugin.java
@@ -36,6 +36,7 @@ import net.runelite.api.Client;
 import net.runelite.api.GameState;
 import net.runelite.api.events.ConfigChanged;
 import net.runelite.api.events.FocusChanged;
+import net.runelite.api.events.GameStateChanged;
 import net.runelite.client.config.ConfigManager;
 import net.runelite.client.eventbus.Subscribe;
 import net.runelite.client.plugins.Plugin;
@@ -93,6 +94,10 @@ public class FpsPlugin extends Plugin
 
 	private final ScheduledExecutorService pingExecutorService = new ExecutorServiceExceptionLogger(Executors.newSingleThreadScheduledExecutor());
 
+	private boolean loaded = false;
+
+	private boolean shutdown;
+
 	@Getter(AccessLevel.PACKAGE)
 	private FpsLimitMode limitMode;
 
@@ -128,6 +133,12 @@ public class FpsPlugin extends Plugin
 		overlay.onFocusChanged(event);
 	}
 
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		shutdown = event.getGameState() != GameState.LOGGED_IN;
+	}
+
 	@Override
 	protected void startUp() throws Exception
 	{
@@ -137,7 +148,13 @@ public class FpsPlugin extends Plugin
 		overlayManager.add(overlay);
 		drawManager.registerEveryFrameListener(drawListener);
 		drawListener.reloadConfig();
-		pingExecutorService.scheduleAtFixedRate(this::getPingToCurrentWorld, 5, 5, TimeUnit.SECONDS);
+		shutdown = client.getGameState() != GameState.LOGGED_IN;
+
+		if (!loaded)
+		{
+			pingExecutorService.scheduleAtFixedRate(this::getPingToCurrentWorld, 5, 5, TimeUnit.SECONDS);
+			loaded = true;
+		}
 	}
 
 	@Override
@@ -145,12 +162,12 @@ public class FpsPlugin extends Plugin
 	{
 		overlayManager.remove(overlay);
 		drawManager.unregisterEveryFrameListener(drawListener);
-		pingExecutorService.shutdown();
+		shutdown = true;
 	}
 
 	private void getPingToCurrentWorld()
 	{
-		if (client.getGameState().equals(GameState.LOGGED_IN) && drawPing)
+		if (!shutdown && drawPing)
 		{
 			ping = Ping.ping(String.format("oldschool%d.runescape.com", client.getWorld() - 300));
 		}

